robottelo/hosts.py

```diff
@@ -1023,12 +1023,14 @@ def configure_puppet(self, proxy_hostname=None):
         self.execute('/opt/puppetlabs/bin/puppet agent -t')
         proxy_host = Host(hostname=proxy_hostname)
         proxy_host.execute(f'puppetserver ca sign --certname {cert_name}')
-        # This particular puppet run would create the host entity under
-        # 'All Hosts' and let's redirect stderr to /dev/null as errors at
-        #  this stage can be ignored.
-        result = self.execute('/opt/puppetlabs/bin/puppet agent -t 2> /dev/null')
-        if result.status:
-            raise ContentHostError('Failed to configure puppet on the content host')
+
+        if run_puppet_agent:
+            # This particular puppet run would create the host entity under
+            # 'All Hosts' and let's redirect stderr to /dev/null as errors at
+            #  this stage can be ignored.
+            result = self.execute('/opt/puppetlabs/bin/puppet agent -t 2> /dev/null')
```

Comment:
Using `-t` isn't quite a good test. While testing the original BZ I noticed it is NOT a reproducer for the issue.

`-t` (or `--test`) is a short hand for multiple options. Quoting the man page:

```
* --test:
  Enable the most common options used for testing. These are 'onetime',
  'verbose', 'no-daemonize', 'no-usecacheonfailure', 'detailed-exitcodes',
  'no-splay', and 'show_diff'.
```

The `--verbose` part always creates entries in the report. If you only want to run puppet it should be sufficient to use `puppet agent --onetime --no-daemonize --no-splay`.